### ISSUE TITLE
Cockpit 'Internal error in login process'

### ISSUE THREAD

**Issue by danielkrajnik** [cockpit-project/cockpit#15645]
Cockpit version: apt latest package [02/04/2021]
OS: Windows Subsystem for Linux 18.04, 20.04

Installation via apt package seemed to be fine, but after the browser login error message pops up "Internal error in login process" on both distros. I assume it must have something to do with PAM modules set up on WSL. 

Steps to reproduce

1. sudo apt install cockpit
2. localhost:9090
3. try to login with username and password

Journald

Apr 02 14:16:08 systemd[1]: Starting Cockpit Web Service...
Apr 02 14:16:08 systemd[1]: Started Cockpit Web Service.
Apr 02 14:16:08 cockpit-ws[14398]: Using certificate: /etc/cockpit/ws-certs.d/0-self-signed.cert
Apr 02 14:16:25 cockpit-ws[14398]: cockpit-session: pam: "System is booting up. See pam_nologin(8)"
Apr 02 14:16:25 cockpit-ws[14398]: cockpit-session: user account access failed: 7 dan: Authentication failure
Apr 02 14:16:25 cockpit-ws[14398]: Received unparseable control message: <data>:1:68: Parse error: invalid empty string constant "", expected valid string constant
Apr 02 14:17:33 systemd[1]: Stopping Cockpit Web Service...
Apr 02 14:17:33 systemd[1]: Stopped Cockpit Web Service.

**Comment by danielkrajnik** [comment]
OS: 18.04 and 20.04

Installation via apt package seems to be successful, but after logining from the browser "Internal error in login process" pops up on both distros

**Comment by marusak** [comment]
`pam: "System is booting up. See pam_nologin(8)"`
This seems to be interesting. Is it still happening or was that happening only right after boot?
If it persists, then maybe pam rules need to be revised.

**Comment by danielkrajnik** [comment]
Thanks for looking into this. I uninstalled Cockpit from WSL instances already, but it was the same problem on two different versions of Ubuntu tested roughly around the same time during a single session/boot (if that makes sense)

**Comment by marusak** [comment]
Do you still have any of those machines to test this stuff out or to check some settings?

**Comment by danielkrajnik** [comment]
yeah, I quickly apt-installed cockpit into ubuntu-20.04 again, but this time the error message is "Internal error in login process"

**Comment by danielkrajnik** [comment]
strace: Process 1144 attached
epoll_wait(6, [{EPOLLIN, {u32=3, u64=3}}], 1, -1) = 1
accept4(3, NULL, NULL, SOCK_CLOEXEC)    = 8
timerfd_settime(7, 0, {it_interval={tv_sec=0, tv_nsec=0}, it_value={tv_sec=0, tv_nsec=0}}, NULL) = 0
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7f0b71b45000
mprotect(0x7f0b71b46000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7f0b72344db0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[1156], tls=0x7f0b72345700, child_tidptr=0x7f0b723459d0) = 1156
epoll_wait(6, [{EPOLLIN, {u32=3, u64=3}}], 1, -1) = 1
accept4(3, NULL, NULL, SOCK_CLOEXEC)    = 9
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7f0b71344000
mprotect(0x7f0b71345000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7f0b71b43db0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[1157], tls=0x7f0b71b44700, child_tidptr=0x7f0b71b449d0) = 1157
epoll_wait(6, [{EPOLLIN, {u32=3, u64=3}}], 1, -1) = 1
accept4(3, NULL, NULL, SOCK_CLOEXEC)    = 12
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7f0b70b43000
mprotect(0x7f0b70b44000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7f0b71342db0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[1161], tls=0x7f0b71343700, child_tidptr=0x7f0b713439d0) = 1161
epoll_wait(6,

**Comment by danielkrajnik** [comment]
strangely enough, strace didn't record anything when I tried to login in to the browser, so I guess that something failed already before that

**Comment by marusak** [comment]
anything in journal during the login process?

**Comment by danielkrajnik** [comment]
no, actually this time there were only two entries:
Apr 05 20:22:22 systemd[1]: Starting Cockpit Web Service...
Apr 05 20:22:22 systemd[1]: Started Cockpit Web Service.

**Comment by danielkrajnik** [comment]
The issue must be with PAM, not Cockpit itself. I also tried to install and login into a jupyterhub, which uses PAM by default. A similar "Invalid username or password" error showed up.

**Comment by Al-Green-COS** [comment]
I encountered this error with configuring cockpit on RHEL 8.6. The fix action I took was to:
1) 'sudo yum remove cockpit',
2) 'sudo yum install cockpit' and 
3) ' sudo systemctl restart cockpit.socket'.  

I then refreshed/reloaded the web browser on my remote system. Upon reload, the login credential and password fields were then visible. Hopefully, this approach benefits others who may encounter this issue.
